docs/rfc/rfc-020-onboarding-projects.rst

```diff
@@ -0,0 +1,235 @@
+collect and process output, and be able to manage entire suites of cases. 
+
+It might also be useful to implement an e2e test orchestrator that runs all
+tendermint instances in a single process, using "real" networks for faster
+feedback and iteration during development.
+
+In addition to being a bit easier to maintain, having a more capable runner
+implementation would make it easier to collect data from test runs, improve
+debugability and reporting. 
+
+Fan-Out For CI E2E Tests
+++++++++++++++++++++++++
+
+While there are some parallelism in the execution of e2e tests, each e2e test
+job must build a tendermint e2e image, which takes about 5 minutes of CPU time
+per-task, which given the size of each of the runs. 
+
+We'd like to be able to reduce the amount of overhead per-e2e tests while
+keeping the cycle time for working with the run a lot of different e2e
+configurations,and also have. While this is in some ways an impossible
+tradeoff, the percentage of overhead at the moment is large enough that we can
+make some material progress with a moderate amount of time. 
+
+Most of this work has to do with modifying github actions configuration and
+e2e artifact (docker) building to reduce redundant work. Eventually, when we
+can drop the requirement for CGo storage engines, it will be possible to move
+(cross) compile tendermint locally, and then inject the binary into the docker
+container, which would reduce a lot of the build-time complexity, although we
+can move more in this direction or have runtime flags to disable CGo
+dependencies for local development.
+
+Remove Panics
+~~~~~~~~~~~~~
+
+There are lots of places in the code base which can panic, and would not be
+particularly well handled. While in some cases, panics are the right answer,
+in many cases the panics were just added to simplify downstream error
+checking, and could easily be converted to errors.
+
+The `Don't Panic RFC
+<https://github.com/tendermint/tendermint/blob/master/docs/rfc/rfc-008-do-not-panic.MD>`_
+covers some of the background and approach.
+
+While the changes are in this project are relatively rote, this will provide
+exposure to lots of different areas of the codebase as well as insight into
+how different areas of the codebase interact with eachother, as well as
+experience with the test suites and infrastructure. 
+
+Implement ABCI an Application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tendermint maintains two very simple ABCI applications (a KV application used
+for basic testing, and slightly more advanced test application used in the
+end-to-end tests). Writing an application would provide a new engineer with
+useful experiences using Tendermint that mirrors the expierence of downstream
+users. 
+
+This is more of an exploratory project, but could include providing common
+interfaces on top of Tendermint consensus for other well known protocols or
+tools (e.g. ``etcd``) or a DNS server or some other tool.
+
+Pauseable Reactors
```

Comment:
"Pauseable" prefigures a particular approach that I'm not sure we know we want. Perhaps reactors should stop, to be resumed by some external signal ("pauseable"), or perhaps they should run continuously and recognize when they are applicable (perhaps "self-regulating").

Either would be reasonable, and I don't have a clear argument (apart from intuition) that one is better. (And there may be other points in the design space I haven't thought about. The clue in the name "reactor" is somewhat suggestive though).

Reply:
I like "self-regulating" as a way to describe this. I think that the design of this is less clear, and that we'd definitely want to make sure someone had thought through a reasonable design before someone started working on this.